EQ FIR optimize proposal

singalsu opened the pull request:
This patch optimizes FIR EQ computation speed a lot when compiled with xt-xcc.

The IPC is updated to match ALSA conventions with binary and enum controls.

Also code style is fixed to pass the pre-commit and post-commit checks.

lgirdwood commented on src/audio/fir_config.h:
```diff
@@ -0,0 +1,76 @@
+/* Select optimized code variant when xt-xcc compiler is used */
+#if FIR_AUTOARCH == 1
+#if defined __XCC__
+//#include <xtensa/config/core-isa.h>
```
needed ?

singalsu replied:
I changed the pull request into next from edit menu but there's a lot of other commits and even conflicts in other parts than my work. It looks like next was behind master.

lgirdwood replied:
@singalsu you will need to rebase on top of next, this will remove other commits

singalsu commented:
@lgirdwood I've now rebased to next and fixed that // comment line. It was a mistake and that header file that I had commented out for some weird reason needs to be included.
